solutions/find_max_in_list.py

```diff
@@ -0,0 +1,49 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+"""
+Created on 2024-12-27
+
+@author: Yuri Spizhovyi
+"""
+
+
+def find_max_in_list(numbers: list[int]) -> int | None:
+    """
+    Find and return the maximum number in a given list.
+
+    Args:
+        numbers (list[int]): A list of integers.
+
+    Returns:
+        int: The maximum number in the list.
+        None: If the list is empty.
+
+    Raises:
+        TypeError: If the input is not a list or if the list contains non-integer elements.
+
+    Tests:
+
+    >>> print(find_max_in_list([]))
+    None
+
+    >>> print(find_max_in_list([0, 0]))
+    0
+
+    >>> print(find_max_in_list([1, 2, 3]))
+    3
+
```

Comment:
You have too many doc string, evan said we should have at most 3-4

Reply:
but I have 5😊 I followed the example given by Evan in the even_odd task (TDD with LLM lesson), where he created 7 doctests. I hope this aligns with the intent of the guidelines. 
Additionally, there are no strict rules about the number of tests—we generally aim for as many tests as necessary to ensure robust functionality. The more tests we have, the better we can cover different scenarios and edge cases. 
Let me know if you'd like to discuss further!